src/cards/contaminate.ts

```diff
@@ -47,8 +48,9 @@ interface CurseData {
 }
 
 // separate function to handle synchronous recursion and animation - avoids long wait times
-async function contaminate(casterPlayer: IPlayer | undefined, unit: IUnit, underworld: Underworld, extraRadius: number, prediction: boolean, quantity: number) {
-  const range = (COLLISION_MESH_RADIUS * 4 + extraRadius) * (casterPlayer?.mageType == 'Witch' ? 1.5 : 1);
+async function contaminate(casterPlayer: IPlayer | undefined, unit: IUnit, quantity: number, radiusBoost: number, underworld: Underworld, prediction: boolean) {
+  const adjustedRadiusBoost = quantity - 1 + radiusBoost;
```

Comment:
why "- 1"?

Reply:
The common formula I'm using is
adjustedRange = baseRange * (1 + (x% * radiusBoost)

or essentially
adjustedRange = baseRange + (some% of baseRange per **_additional_** quantity/plus radius)

The range of a spell doesn't usually double for every quantity/plus radius, so we count the initial quantity and the "extra" quantities separate, hence the - 1. The first quantity will give the entire base range, and is handled in the "baseRange * 1" part of the equation, while later quantities and "plus radius" only give a portion of that range, and are handled in the "+ (0.x * radiusBoost) part of the equation, where radiusBoost usually = (_extra_ quantity + state.aggregator.radiusBoost)

Reply:
All this being said, Contaminate's radius isn't supposed to scale with quantity anyways, so I fixed that
(Contaminate is a special case where extra quantity causes chaining instead of increased aoe radius: Increased radius will come from "Plus Radius" and the Witch class only.)